lib/cocoapods/target/pod_target.rb

```diff
@@ -150,7 +150,11 @@ def all_files
     # @return [Pathname] the pathname for headers in the sandbox.
     #
     def headers_sandbox
-      Pathname.new(product_module_name)
+      if root_spec.attributes_hash.key?('header_dir')
```

Comment:
we might want something a bit more robust for this in the future (maybe `root_spec.consumer(platform).header_dir`?) but this should work for now

Reply:
yes can we use `consumer` here?

Reply:
My first attempt to use `consumer` generated a bunch of test failures. I'll make another attempt to work though them later on today.

Reply:
Why is using `consumer` more robust?

Reply:
My thought was it would remove the reliance on knowing the structure of attributes_hash, but I can see how that’s not really a problem in practice

Reply:
I suppose it's at least a consistent style to use `consumer`. I made the update.